polaris-react/src/components/Text/Text.tsx

```diff
@@ -43,9 +45,9 @@ const VariantFontWeightMapping: {[V in Variant]: FontWeight} = {
   heading2xl: 'semibold',
   heading3xl: 'semibold',
   heading4xl: 'bold',
-  bodySm: 'regular',
-  bodyMd: 'regular',
-  bodyLg: 'regular',
+  bodyLg: null,
+  bodyMd: null,
+  bodySm: null,
```

Comment:
Setting these to `null` and not adding a `font-weight` if `null` is returned. This means less CSS should be rendered for the majority of body components.